docs/contributors/code/coding-guidelines.md

```diff
@@ -136,10 +136,306 @@ export { __unstableDoTerribleAwfulAction } from './api';
 -   An **experimental API** is one which is planned for eventual public availability, but is subject to further experimentation, testing, and discussion.
 -   An **unstable API** is one which serves as a means to an end. It is not desired to ever be converted into a public API.
 
-In both cases, the API should be made stable or removed at the earliest opportunity.
+In both cases, the API should be made stable or removed at the earliest opportunity. 
 
 While an experimental API may often stabilize into a publicly-available API, there is no guarantee that it will. The conversion to a stable API will inherently be considered a breaking change by the mere fact that the function name must be changed to remove the `__experimental` prefix.
 
+#### Experimental APIs merged into WordPress Core become a liability
+
+**Avoid introducing public experimental APIs.**
+
+As of June 2022, WordPress Core contains 280 publicly exported experimental APIs. They got merged from the Gutenberg
+plugin during the major WordPress releases. Many plugins and themes rely on these experimental APIs for essential
+features that can't be accessed in any other way. Naturally, these APIs can't be removed without a warning anymore.
+They are a part of the WordPress public API and fall under the 
+[WordPress Backwards Compatibility policy](https://developer.wordpress.org/block-editor/contributors/code/backward-compatibility/). 
+Removing them involves a deprecation process that spans multiple WordPress releases. 
```

Comment:
I think the length really depends on the API, some are not useful or not used at all by plugins and should be easier to remove, others are less so.

Reply:
Good note, I rewrote this to note both extremes (fairly straightforward vs hard and long)